File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/DyedGroup.java
package generations.gg.generations.core.generationscore.common.world.level.block;

import net.minecraft.core.Holder;
import net.minecraft.world.item.DyeColor;
import net.minecraft.world.level.block.Block;

import java.util.Map;

public record DyedGroup(Map<DyeColor, Holder<Block>> block)  {
    public Block[] toArray() {
        return block().values().stream().map(Holder::value).map(a -> (Block) a).toArray(Block[]::new);
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/DyedGroup.kt
package generations.gg.generations.core.generationscore.common.world.level.block

import net.minecraft.core.Holder
import net.minecraft.world.item.DyeColor
import net.minecraft.world.level.block.Block

@JvmRecord
data class DyedGroup(@JvmField val block: Map<DyeColor, Holder<out Block>>) {
    fun toArray(): Array<Holder<out Block>> = block.values.toTypedArray()
    fun toExposedArray(): Array<out Block> = block.values.map(Holder<out Block>::value).toTypedArray()
}
